src/ipc/handler.c

```diff
@@ -738,6 +738,30 @@ static int ipc_dma_trace_config(uint32_t header)
 	return err;
 }
 
+static int ipc_dma_trace_filter_update(uint32_t header)
+{
+	struct ipc *ipc = ipc_get();
+	struct sof_ipc_dma_trace_filter_elem *elem;
+	struct sof_ipc_dma_trace_filter *packet = ipc->comp_data;
+	int ret;
+	int i;
+
+	tr_info(&ipc_tr, "ipc: trace_filter_update received, size %d elems",
+		packet->elem_cnt);
+
+	for (i = 0; i < packet->elem_cnt; ++i) {
+		elem = &packet->elems[i];
+		ret = trace_filter_update(elem->uuid_key, elem->log_level);
+		if (ret) {
+			tr_info(&ipc_tr, "ipc: trace_filter_update error for UUID key 0x%X and log level %d",
+				elem->uuid_key, elem->log_level);
+			return ret;
```

Comment:
This should really keep going, but return the error status at the end. This way at least the user has most of the trace setting updated. Please also add comments to this functions.